### ISSUE TITLE
Local variables are broken by mrb_fiber_resume

### ISSUE THREAD

**Issue by charliesome** [mruby/mruby#5287]
```c
#include <mruby.h>
#include <mruby/compile.h>
#include <mruby/string.h>
#include <stdio.h>

mrb_state* mrb;

void p(mrb_value value) {
    printf("--> %s\n", mrb_str_to_cstr(mrb, mrb_inspect(mrb, value)));
}

int main() {
    mrb = mrb_open();

    mrb_value fiber_class = mrb_obj_value(mrb_class_get(mrb, "Fiber"));

    mrb_value proc = mrb_load_string(mrb,
        "proc do\n"
        "  ary = [1,2,3]\n"
        "  while elem = ary.pop\n"
        "    p ary\n"
        "    Fiber.yield(elem)\n"
        "    p ary\n"
        "  end\n"
        "end\n"
    );

    mrb_gc_protect(mrb, proc);

    mrb_sym new = mrb_intern_cstr(mrb, "new");
    mrb_value fiber = mrb_funcall_with_block(mrb, fiber_class, new, 0, NULL, proc);
    mrb_gc_protect(mrb, fiber);

    p(mrb_fiber_resume(mrb, fiber, 0, NULL));
    p(mrb_fiber_resume(mrb, fiber, 0, NULL));
    p(mrb_fiber_resume(mrb, fiber, 0, NULL));
    p(mrb_fiber_resume(mrb, fiber, 0, NULL));
    p(mrb_fiber_resume(mrb, fiber, 0, NULL));
}
```

Expected output:

```
[1, 2]
--> 3
[1, 2]
[1]
--> 2
[1]
[]
--> 1
[]
--> nil
--> resuming dead fiber (FiberError)
```

Actual output:

```
[1, 2]
--> 3
nil
--> undefined method 'pop' (NoMethodError)
resuming dead fiber (FiberError)
```

**Comment by charliesome** [comment]
The trampoline workaround described in https://github.com/mruby/mruby/issues/5261#issuecomment-756593677 behaves correctly in this case.

**Comment by dearblue** [comment]
When I tried it on #5272, I got the expected result.
With the current `mrb_callinfo` structure, it probably requires saving and reverting stack (similar to commit a0c1e075e35c358d21934c28ff1bec4153502409).

**Comment by matz** [comment]
I merged #5272 that should fix this issue too. Thank you @dearblue for fixing it. Thanks, @charliesome for reporting.
